purchase_order_duplicate_check/models/purchase_order.py

```diff
@@ -0,0 +1,41 @@
+from odoo import models, tools
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _prepare_pending_orders_message(self, product_id):
+        """
+        Prepare pending order line message
+
+        :param product_id: product.product record id
+        :return str: message
+        """
+        message_parts = []
+        order_lines = self.order_line.filtered(lambda l: l.product_id.id == product_id)
+        for line in order_lines:
+            order = line.order_id
+            order_href = (
+                f"<a href='/web#id={order.id}&model={order._name}'>{order.name}</a>"
+            )
+            type_ = order.state in ["draft", "sent"] and "RFQ" or "PO"
+            message_parts.append(
+                f"{type_}: {order_href}; date: {order.create_date.date()}; Qty: {line.product_qty};<br/>"  # noqa
+            )
+        return "".join(message_parts)
+
+    def _check_pending_order(self):
+        """Check for pending orders and trigger confirmation wizard if needed."""
+        if not tools.config["test_enable"]:
+            action = self.env["confirmation.wizard"].confirm_pending_order(self)
+            if action:
+                return action
```

Comment:
Adding the test_enable will allow other modules to be tested properly, usually, you can see in many places of Odoo code after adding that a context is checked. So maybe you can avoid changing the config['test_enable'] to True to run your test just adding the following context to the test.

```suggestion
    def _check_pending_order(self):
        """Check for pending orders and trigger confirmation wizard if needed."""
        if not tools.config["test_enable"] or self._context.get('test_purchase_duplicate_check') :
            action = self.env["confirmation.wizard"].confirm_pending_order(self)
            if action:
                return action
        return {}        
```